src/util/file_watch.c

```diff
@@ -131,15 +134,32 @@ static int watched_file_inotify_cb(const char *filename,
         return EINVAL;
     }
 
-    if (strcmp(fw_ctx->filename, filename) == 0) {
-        if (access(fw_ctx->filename, F_OK) == 0) {
-            fw_ctx->cb(fw_ctx->filename, fw_ctx->cb_arg);
+    res = realpath(fw_ctx->filename, expected);
+    if (res == NULL) {
+         DEBUG(SSSDBG_TRACE_LIBS,
+               "Normalization failed for %s is missing. Skipping the callback.\n",
```

Comment:
I can't parse this string... Should "is missing" be just removed?
It would be also good to use different wording for 'expected' and 'received' to be able to distinguish those logs messages easily.

Reply:
You are right. That is what happens when you copy, paste and update the line.